crates/aqd-polkadot/src/commands/call.rs

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use {
+    anyhow::{anyhow, Context, Result},
+    colored::Colorize,
+    serde_json::{json, to_string_pretty},
+    std::{fmt::Debug, process::exit},
+};
+
+use {
+    super::CLIExtrinsicOpts,
+    aqd_utils::{
+        check_target_match, print_key_value, print_title, print_warning, prompt_confirm_transaction,
+    },
+    contract_build::Verbosity,
+    contract_extrinsics::{
+        BalanceVariant, CallCommandBuilder, DefaultConfig, ExtrinsicOptsBuilder, StorageDeposit,
+    },
+    subxt::Config,
+};
+
+#[derive(Debug, clap::Args)]
+#[clap(name = "call", about = "Call a contract on Polkadot")]
+pub struct PolkadotCallCommand {
+    #[clap(
+        name = "contract",
+        long,
+        help = "Specifies the address of the contract to call."
+    )]
+    contract: <DefaultConfig as Config>::AccountId,
+    #[clap(
+        long,
+        short,
+        help = "Specifies the name of the contract message to call."
+    )]
+    message: String,
+    #[clap(long, num_args = 0.., help = "Specifies the arguments of the contract message to call.")]
+    args: Vec<String>,
+    #[clap(flatten)]
+    extrinsic_cli_opts: CLIExtrinsicOpts,
+    #[clap(
+        name = "value",
+        long,
+        default_value = "0",
+        help = "Specifies the value to be transferred as part of the call."
+    )]
+    value: BalanceVariant,
+    #[clap(
+        name = "gas",
+        long,
+        help = "Specifies the maximum amount of gas to be used for this command."
+    )]
+    gas_limit: Option<u64>,
+    #[clap(long, help = "Specifies the maximum proof size for this call.")]
+    proof_size: Option<u64>,
+    #[clap(
+        short('y'),
+        long,
+        help = "Specifies whether to skip the confirmation prompt."
+    )]
+    skip_confirm: bool,
+}
+
+impl PolkadotCallCommand {
+    /// Returns whether to export the call output in JSON format.
+    pub fn output_json(&self) -> bool {
+        self.extrinsic_cli_opts.output_json
+    }
+
+    /// Handles the calling of a contract on the Polkadot network.
+    ///
+    /// If the `execute` flag is set to `false`, it performs a dry run of the call and displays
+    /// the results. If the `output_json` flag is set to `true`, the output is in JSON format.
+    /// Otherwise, it prompts for a transaction confirmation and then submits the transaction for execution.
+    pub async fn handle(&self) -> Result<()> {
```

Comment:
There are "read only" messages in contracts. They'll have the `mutates` flag set to false in the metadata. For those messages, the `-x/--execute` should lead to an error instead.